### PR TITLE
DAGMC void material assignment fix

### PR DESCRIPTION
**Pull request by pshriwise**
This addresses #1372, raised by @Shimwell (thanks!)

Material names with the "graveyard", "vacuum", and "void"  are used to indicate void material assignment, but any name *containing* those keywords were being assigned as such incorrectly (e.g. `vacuum_vessel`).

This PR fixes that oversight and updates material names in some of the DAGMC tests to verify that this change was successful. The material names have been updated while the test results remain the same, indicating that the new material name containing "void" is no longer misidentified as a void region.

I've also updated the DAGMC legacy to a proper test class. Something may have slipped through the cracks there as well.

### CODE REVIEW

File: src/dagmc.cpp [hunk 1]
```diff
@@ -268,14 +268,12 @@ void load_dagmc_geometry()
     std::string cmp_str = mat_value;
     to_lower(cmp_str);
 
-    if (cmp_str.find("graveyard") != std::string::npos) {
+    if (cmp_str == "graveyard") {
```
Comment (makeclean) [inline src/dagmc.cpp]: dumb question is cmp_str to_lower() at some point? so Graveyard and Vacuum would count too?

**Comment by Shimwell** [comment]
I have run this branch with the model that was previously breaking and a few other DAGMC models. It works for me and I am now once again getting neutrons interacting with my vacuum_vessel. Nice fix @pshriwise

File: src/dagmc.cpp (hunk 1 above)
Comment (Shimwell) [reply src/dagmc.cpp]: I have tried Vacuum_vessel and vacuum_vessel and both cases work fine for me.

File: tests/regression_tests/dagmc/legacy/test.py [hunk 2]
```diff
@@ -8,42 +8,48 @@
     not openmc.lib._dagmc_enabled(),
     reason="DAGMC CAD geometry is not enabled.")
 
-def test_dagmc():
-    model = openmc.model.Model()
+class DAGMCLegacyTest(PyAPITestHarness):
+
+    def _build_inputs(self):
```
Comment (paulromano) [inline tests/regression_tests/dagmc/legacy/test.py]: You actually don't need to subclass `PyAPITestHarness` just to change the model. You can pass a `Model` object as the second argument when instantiating `PyAPITestHarness(...)`. See the `void` regression test as an example.

**Pull request merged**